scubagoggles/rego/Drive.rego

```diff
@@ -0,0 +1,1588 @@
+
+LogMessage1_8 := "DEFAULT_LINK_SHARING_FOR_NEW_DOCS"
+
+Check1_8_OK if {
+    not PolicyApiInUse
+    events := utils.FilterEventsOU(LogEvents, LogMessage1_8, utils.TopLevelOU)
+    count(events) > 0
+}
+
+Check1_8_OK if {PolicyApiInUse}
+
+NonComplianceMessage1_8(value) := sprintf("When users create items, the default access is set to: %s",
+                                          [value])
+
+GetFriendlyValue1_8(Value):= "private to the owner." if {
+    Value in {"PRIVATE", "PRIVATE_TO_OWNER"}
+} else := "the primary target audience can access the item if they have the link" if {
+    Value in {"PEOPLE_WITH_LINK", "PRIMARY_AUDIENCE_WITH_LINK"}
+} else := "the primary target audience can search and find the item." if {
+    Value in {"PUBLIC", "PRIMARY_AUDIENCE_WITH_LINK_OR_SEARCH"}
+} else := Value
+
+NonCompliantOUs1_8 contains {
+    "Name": OU,
+    "Value": NonComplianceMessage1_8(GetFriendlyValue1_8(LastEvent.NewValue))
+}
+if {
+    not PolicyApiInUse
+    some OU in utils.OUsWithEvents
+    Events := utils.FilterEventsOU(LogEvents, LogMessage1_8, OU)
+    count(Events) > 0
+    LastEvent := utils.GetLastEvent(Events)
+    LastEvent.NewValue != "PRIVATE"
+    LastEvent.NewValue != "INHERIT_FROM_PARENT"
+}
+
+NonCompliantGroups1_8 contains {
+    "Name": Group,
+    "Value": NonComplianceMessage1_8(GetFriendlyValue1_8(LastEvent.NewValue))
+}
+if {
+    not PolicyApiInUse
+    some Group in utils.GroupsWithEvents
+    Events := utils.FilterEventsGroup(LogEvents, LogMessage1_8, Group)
+    count(Events) > 0
+    LastEvent := utils.GetLastEvent(Events)
+    LastEvent.NewValue != "PRIVATE"
+    LastEvent.NewValue != "INHERIT_FROM_PARENT"
+}
+
+NonCompliantOUs1_8 contains {
+    "Name": OU,
+    "Value": NonComplianceMessage1_8(GetFriendlyValue1_8(defaultAccess))
+}
+if {
+    some OU, settings in input.policies
+    DriveEnabled(OU)
+    defaultAccess := settings.drive_and_docs_general_access_default.defaultFileAccess
+    defaultAccess != "PRIVATE_TO_OWNER"
+}
+
+tests contains {
+    "PolicyId": DriveId1_8,
+    "Criticality": "Shall",
+    "ReportDetails": utils.NoSuchEventDetails(DefaultSafe, utils.TopLevelOU),
+    "ActualValue": "No relevant event for the top-level OU in the current logs",
+    "RequirementMet": DefaultSafe,
+    "NoSuchEvent": true
+}
+if {
+    not PolicyApiInUse
+    DefaultSafe := true
+    not Check1_8_OK
+}
+
+tests contains {
+    "PolicyId": DriveId1_8,
+    "Criticality": "Shall",
+    "ReportDetails": utils.ReportDetails(NonCompliantOUs1_8, NonCompliantGroups1_8),
+    "ActualValue": {"NonCompliantOUs": NonCompliantOUs1_8, "NonCompliantGroups": NonCompliantGroups1_8},
+    "RequirementMet": Status,
+    "NoSuchEvent": false
+}
+if {
+    Check1_8_OK
+    Conditions := {count(NonCompliantOUs1_8) == 0, count(NonCompliantGroups1_8) == 0}
+    Status := (false in Conditions) == false
+}
+#--
+
+###################
+# GWS.DRIVEDOCS.2 #
+###################
+
+#
+# Baseline GWS.DRIVEDOCS.2.1
+#--
```

Comment:
The report fails you no matter what you select for "Allow members with manager access to override the settings below"

Reply:
![image](https://github.com/user-attachments/assets/f02f00ca-a693-4f2f-95e3-d61ce68fba36)

Reply:
After implementing Google's policy "reduction" process, this is working for me
when I change settings in the Admin UI.